### ISSUE TITLE
jobs:work: Octokit::TooManyRequests: GET https://api.github.com/users/concerto-addons/repos: 403 - API rate limit exceeded

### ISSUE THREAD

**Issue by salfter** [concerto/concerto#1238]
The script that updates RSS feeds keeps crashing.  It keeps trying to access GitHub, but runs up against a rate limit.  It says that authenticated requests get a higher limit, but AFAIK there's no information in the Concerto documentation on setting authentication parameters.

Here's the trace output:

```
** Invoke jobs:work (first_time)
** Invoke jobs:environment_options (first_time)
** Invoke environment (first_time)
** Execute environment
rake aborted!
Octokit::TooManyRequests: GET https://api.github.com/users/concerto-addons/repos: 403 - API rate limit exceeded for 24.120.210.226. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) // See: https://developer.github.com/v3/#rate-limiting
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/octokit-3.8.0/lib/octokit/response/raise_error.rb:16:in `on_complete'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/faraday-0.9.1/lib/faraday/response.rb:9:in `block in call'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/faraday-0.9.1/lib/faraday/response.rb:57:in `on_complete'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/faraday-0.9.1/lib/faraday/response.rb:8:in `call'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/faraday-0.9.1/lib/faraday/rack_builder.rb:139:in `build_response'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/faraday-0.9.1/lib/faraday/connection.rb:377:in `run_request'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/faraday-0.9.1/lib/faraday/connection.rb:140:in `get'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/sawyer-0.6.0/lib/sawyer/agent.rb:94:in `call'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/octokit-3.8.0/lib/octokit/client.rb:339:in `request'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/octokit-3.8.0/lib/octokit/client.rb:200:in `paginate'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/octokit-3.8.0/lib/octokit/client/repositories.rb:68:in `repositories'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/octokit-3.8.0/lib/octokit.rb:27:in `method_missing'
/var/www/concerto/app/models/concerto_plugin.rb:22:in `concerto_addons'
/var/www/concerto/app/models/concerto_plugin.rb:30:in `<class:ConcertoPlugin>'
/var/www/concerto/app/models/concerto_plugin.rb:12:in `<top (required)>'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:443:in `load'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:443:in `block in load_file'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:633:in `new_constants_in'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:442:in `load_file'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:342:in `require_or_load'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:480:in `load_missing_constant'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:180:in `const_missing'
/var/www/concerto/config/initializers/12-plugins.rb:8:in `<top (required)>'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:241:in `load'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:241:in `block in load'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:232:in `load_dependency'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:241:in `load'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/engine.rb:648:in `block in load_config_initializer'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/notifications.rb:161:in `instrument'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/engine.rb:647:in `load_config_initializer'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/engine.rb:611:in `each'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/engine.rb:611:in `block in <class:Engine>'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/initializable.rb:30:in `instance_exec'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/initializable.rb:30:in `run'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/initializable.rb:55:in `block in run_initializers'
/usr/lib64/ruby/1.9.1/tsort.rb:150:in `block in tsort_each'
/usr/lib64/ruby/1.9.1/tsort.rb:183:in `block (2 levels) in each_strongly_connected_component'
/usr/lib64/ruby/1.9.1/tsort.rb:210:in `block (2 levels) in each_strongly_connected_component_from'
/usr/lib64/ruby/1.9.1/tsort.rb:219:in `each_strongly_connected_component_from'
/usr/lib64/ruby/1.9.1/tsort.rb:209:in `block in each_strongly_connected_component_from'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/initializable.rb:44:in `each'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/initializable.rb:44:in `tsort_each_child'
/usr/lib64/ruby/1.9.1/tsort.rb:203:in `each_strongly_connected_component_from'
/usr/lib64/ruby/1.9.1/tsort.rb:182:in `block in each_strongly_connected_component'
/usr/lib64/ruby/1.9.1/tsort.rb:180:in `each'
/usr/lib64/ruby/1.9.1/tsort.rb:180:in `each_strongly_connected_component'
/usr/lib64/ruby/1.9.1/tsort.rb:148:in `tsort_each'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/initializable.rb:54:in `run_initializers'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/application.rb:300:in `initialize!'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/railtie.rb:194:in `public_send'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/railtie.rb:194:in `method_missing'
/var/www/concerto/config/environment.rb:10:in `<top (required)>'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:247:in `require'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:247:in `block in require'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:232:in `load_dependency'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/activesupport-4.1.12/lib/active_support/dependencies.rb:247:in `require'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/application.rb:276:in `require_environment!'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/railties-4.1.12/lib/rails/application.rb:405:in `block in run_tasks_blocks'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:240:in `call'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:240:in `block in execute'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:235:in `each'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:235:in `execute'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/usr/lib64/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:199:in `each'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:199:in `invoke_prerequisites'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/usr/lib64/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:199:in `each'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:199:in `invoke_prerequisites'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/usr/lib64/ruby/1.9.1/monitor.rb:211:in `mon_synchronize'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/task.rb:165:in `invoke'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:150:in `invoke_task'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `each'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:106:in `block in top_level'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:115:in `run_with_threads'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:100:in `top_level'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:78:in `block in run'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:176:in `standard_exception_handling'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/lib/rake/application.rb:75:in `run'
/var/www/concerto/vendor/bundle/ruby/1.9.1/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/var/www/concerto/vendor/bundle/ruby/1.9.1/bin/rake:23:in `load'
/var/www/concerto/vendor/bundle/ruby/1.9.1/bin/rake:23:in `<main>'
Tasks: TOP => jobs:work => jobs:environment_options => environment
```

**Comment by salfter** [comment]
(If it makes a difference, I'm on commit dfeccb342c7845e5c6e8db0ff101c426c9407790.)

**Comment by augustf** [comment]
@salfter The problem here seems to be twofold. One is that the worker script shouldn't crash for really anything short of the entire application breaking. More immediate to your concern is that the worker script (and other tasks) all end up calling the Github API -- blowing through the draconian limits they have. 

I'll be doing a commit to prevent script invocations from calling on the Github API, but you can also do something in config/initializers with Octokit like this:

```
Octokit.configure do |c|
  c.login = 'defunkt'
  c.password = 'c0d3b4ssssss!'
end
```

**Comment by salfter** [comment]
Just a quick followup: ended up switching to the Debian image instead of trying to get things running reliably on Gentoo.  2.2.7 worked without issue from the Debian image, and once I edited the repo location and installed PostgreSQL libs/headers (unstated dependency?), the image updated to 2.3.0 without issue.  Will still need to be careful with upgrades, but that's why VMware has snapshots. :-)

**Comment by augustf** [comment]
@salfter Thanks for pointing out the postgres bit. That's a change that replaced a scheme we had to detect database engines that Bundler wouldn't accommodate. I've fixed that situation in https://github.com/concerto/concerto-debian/commit/446db827e29c7b3b6ecc79f3c06812011faea501 

In the future, we'd also have an interest in preparing the Concerto package in a more streamlined way 
and for multiple platforms, since our workflow right now is quite focused on Debian and is already quite a development burden.
